client/src/components/DataFiles/DataFilesBreadcrumbs/DataFilesBreadcrumbs.jsx

```diff
@@ -155,52 +176,33 @@ const DataFilesBreadcrumbs = ({
       }
     }, '');
 
+  const fullPath = paths.slice(-1);
+  const currentDirectory = pathComps.slice(-1);
+
   return (
-    <div className={`breadcrumbs ${className}`}>
-      {scheme === 'projects' && (
-        <>
-          <RootProjectsLink
-            api={api}
-            section={section}
-            operation={operation}
-            label="Shared Workspaces"
-          />{' '}
-          {system && `/ `}
-        </>
+    <div className="breadcrumb-container">
+      <div className={`breadcrumbs ${className}`}>
+        {currentDirectory.length === 0 ? (
+          <span className="system-name">
+            {truncateMiddle(systemName || 'Shared Workspaces', 30)}
+          </span>
+        ) : (
+          currentDirectory.map((pathComp, i) => {
+            if (i === fullPath.length - 1) {
+              return <span key={uuidv4()}>{truncateMiddle(pathComp, 30)}</span>;
+            }
+          })
+        )}
+      </div>
+      {systemName && api !== 'googledrive' ? (
```

Comment:
Let's base this on whether or not the api is tapis, as I think that's the only case that works here
```suggestion
      {systemName && api === 'tapis' ? (
```